www_src/pages/project-settings/project-settings.jsx

```diff
@@ -2,79 +2,103 @@ var React = require('react/addons');
 
 var api = require('../../lib/api');
 var render = require('../../lib/render.jsx');
-var router = require('../../lib/router');
 var TextInput = require('../../components/text-input/text-input.jsx');
+var Loading = require('../../components/loading/loading.jsx');
 
 var ProjectSettings = React.createClass({
+
   mixins: [
     React.addons.LinkedStateMixin,
-    router
+    require('../../lib/router')
   ],
+
   getInitialState: function () {
     return {
-      title: ''
+      title: '',
+      loading: false
     };
   },
+
+  uri: function () {
+    return '/users/' + this.state.params.user + '/projects/' + this.state.params.project;
+  },
+
   componentWillMount: function () {
-    var _this = this;
 
-    // Build up URI for API requests
-    var uri = '/users/' + this.state.params.user + '/projects/' + this.state.params.project;
+    this.load();
+
+    // Set up the save method when we press the back button
+    this.props.update({
```

Comment:
This overrides the back  button method on the root component `to call this.save`